### PR TITLE
Set isDirty to true when changing parameter types

### PR DESCRIPTION
**Pull request by rauchy**
feb-17-2019 12-09-40

### CODE REVIEW

File: client/app/components/parameter-settings.html [hunk 1]
```diff
@@ -23,7 +23,7 @@ <h4 class="modal-title">{{ $ctrl.isNewParameter ? 'Add Parameter' : $ctrl.parame
       </div>
       <div class="form-group">
         <label>Type</label>
-        <select ng-model="$ctrl.parameter.type" class="form-control">
+        <select ng-model="$ctrl.parameter.type" class="form-control" ng-change="$ctrl.updateType()">
```
Comment (arikfr) [inline client/app/components/parameter-settings.html]: We need to set isDirty for any value change, not only type. While the type change is important for validation, the rest are important for consistent UI/UX.

**Comment by rauchy** [comment]
> An alternative would be to only use the OK to apply the changes, signal to query that it's dirty but not save (send HTTP request).

I tend to favor this alternative. The whole "parameters dialog has a mind of its own and decides when to save" feels unreliable.

@arik @ranbena should we take advantage of the fact that we need a major change in this component and rewrite it in React?

**Comment by arikfr** [comment]
> should we take advantage of the fact that we need a major change in this component and rewrite it in React?

Of course :-)

**Comment by ranbena** [comment]
Yeah no doubt should add an "ok/cancel" which auto-saves.
@rauchy feel free to assign me and I'll reactify it.

**Comment by rauchy** [comment]
@ranbena Why do you think auto-save makes more sense here? I feel that having things that persist from different places only adds confusion and it's better to have one Save button.

**Comment by ranbena** [comment]
> @ranbena Why do you think auto-save makes more sense here? I feel that having things that persist from different places only adds confusion and it's better to have one Save button.

Because of the visual connotation.
In a conventional form, for instance, it is obvious that a change to any field would require a "save" at the bottom of the form. There's a visual beginning (top of form) that flows downward to an end (submit button).

Here though, the same convention suggests a correlation between the query and the "save" button, but any other field in the page is unassociated.

**Comment by ranbena** [comment]
Discussion in https://discuss.redash.io/t/query-unsaved-changes-indication/3302
Implementation in #3488.

**Pull request closed without merging**